community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/logical/PlanSingleQuery.scala

```diff
@@ -72,6 +72,7 @@ case class PlanSingleQuery(planPart: (PlannerQuery, LogicalPlanningContext, Opti
           leaves.size > 1 && leaves.drop(1).exists(
             nodeOverlap(_, plannerQuery) ||
               relationshipOverlap(plannerQuery) ||
+              setLabelOverlap(plannerQuery) ||
```

Comment:
Same here can this overlap not be handled by the nodeOverlap and just update the return of `labels` in `UpdateGraph`?